conda_build/environ.py

```diff
@@ -469,8 +469,9 @@ def linux_vars(compiler_vars, prefix, config):
     if config.arch == 32:
         compiler_vars['CFLAGS'] += ' -m32'
         compiler_vars['CXXFLAGS'] += ' -m32'
-    return {}
```

Comment:
Yeah, it works, but we're both modifying compiler_vars in place, and then also doing the update?  That seems confusing and fragile.  We should do all of one or the other - set CFLAGS/CXXFLAGS in the return dictionary here (no modification in place) or set everything on compiler vars, and remove the update call.